h/views/admin/groups.py

```diff
@@ -175,7 +181,7 @@ def update(self):
         def on_success(appstruct):
             """Update the group resource on successful form validation."""
 
-            organization = self.organizations[appstruct["organization"]]
+            organization = self.organizations.get(appstruct["organization"])
```

Comment:
This is when updating an existing group as opposed to creating a new one. If the group has no organization then `appstruct["organization"]` will be `None` and `organization` will also be `None`